### PR TITLE
.NET docs updates

### PR DESCRIPTION
**Pull request by PeloWriter**
This issue tracks the work for this PR: https://github.com/getsentry/sentry-docs/issues/3515.

### CODE REVIEW

File: src/includes/getting-started-config/dotnet.mdx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+```csharp {filename:Program.cs}
 using (SentrySdk.Init("___PUBLIC_DSN___"))
 {
     // App code
+
+    // Set TracesSampleRate to 1.0 to capture 100%
+    // of transactions for performance monitoring.
+    // We recommend adjusting this value in production
+    options.TracesSampleRate = 1.0;
 }
```
Comment (bruno-garcia) [inline src/includes/getting-started-config/dotnet.mdx:12]: Once we agree on the snippet (comments etc) we can get the F# version:

```suggestion
using (SentrySdk.Init(o => {
    // Tells which project in Sentry to send events to:
    o => o.Dsn = "___PUBLIC_DSN___";
    // Set TracesSampleRate to 1.0 to capture 100%
    // of transactions for performance monitoring.
    // We recommend adjusting this value in production
    options.TracesSampleRate = 1.0; }))
{
    // App code goes here - Disposing will flush events out
}
```

File: src/includes/getting-started-install/dotnet.mdx [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
-Install-Package Sentry
+```powershell {tabTitle:Package Manager}
+Install-Package Sentry -Version {{ packages.version('sentry.dotnet') }}
+```
 
-# Or using .NET Core CLI
-dotnet add package Sentry
+```shell {tabTitle:.NET Core CLI}
+dotnet add package Sentry -v {{ packages.version('sentry.dotnet') }}
+```
```
Comment (bruno-garcia) [inline src/includes/getting-started-install/dotnet.mdx:9]: Can we invert these two please? Lets have `dotnet add` first from now one

File: src/includes/getting-started-install/dotnet.mdx [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
-# Or using Paket
-paket add Sentry
+```shell {tabTitle:Paket CLI}
+paket add Sentry --version {{ packages.version('sentry.dotnet') }}
 ```
 
-**Using .NET Framework prior to 4.6.1?**
+<Note>
+
+You can <PlatformLink to="/usage/#ignoring-errors">ignore exceptions</PlatformLink> by their type when initializing the SDK.
```
Comment (bruno-garcia) [inline src/includes/getting-started-install/dotnet.mdx:17]: Not sure it's worth adding this in the getting started. Makes sense to have it in the "Configuration" section.
Users won't need this information when they are first trying out the SDK.

File: src/includes/getting-started-primer/dotnet.mdx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+Our .NET SDK supports all recent versions of the platform, and integrates well with a variety of popular frameworks and packages in the ecosystem. It gives developers helpful hints for where and why an error or performance issue might have occurred.
+
+The main [Sentry NuGet package](https://www.nuget.org/packages/Sentry) targets .NET Standard 2.0. [Our legacy SDK](/clients/csharp/) supports .NET Framework as early as 3.5. Read more about [compatibility and unit testing](./unit-testing.mdx).
```
Comment (bruno-garcia) [inline src/includes/getting-started-primer/dotnet.mdx:3]: I think here we could say that:

> The lowest version this SDK is compatible with are .NET Standard 2.0 and .NET Framework 4.6.1.
> For older versions, such as .NET Framework 3.5, please refer to our Raven SDK.

The reference about Unit Testing here isn't a good fit I believe. Also something we can just move to configuration somehow.

File: src/includes/getting-started-primer/dotnet.mdx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+**Features:**
+
+- <PlatformLink to="/enriching-events/breadcrumbs/#automatic-breadcrumbs">Breadcrumbs automatically</PlatformLink> captured
```
Comment (bruno-garcia) [inline src/includes/getting-started-primer/dotnet.mdx:7]: To really clear, is it worth pointing out something like:
```suggestion
- Multiple integrations with additional features such as Serilog, log4net, ASP.NET Core, EntityFramework, Xamarin.
- <PlatformLink to="/enriching-events/breadcrumbs/#automatic-breadcrumbs">Breadcrumbs automatically</PlatformLink> captured
```

?

File: src/includes/getting-started-primer/dotnet.mdx [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+- <PlatformLink to="/enriching-events/attachments/">Attachments</PlatformLink> enrich your event by storing additional files, such as config or log files.
+- <PlatformLink to="/enriching-events/user-feedback/">User Feedback</PlatformLink> provides the ability to collect user information when an event occurs.
+- [Performance Monitoring](/product/performance/) creates transactions to track the performance of your application
+- Samples provided for [C#](https://github.com/getsentry/sentry-dotnet/tree/main/samples), [F# Console](https://github.com/sentry-demos/fsharp), and[F# Giraffe](https://github.com/sentry-demos/giraffe)
```
Comment (bruno-garcia) [inline src/includes/getting-started-primer/dotnet.mdx:11]:
```suggestion
- Samples provided for [multiple C# app models](https://github.com/getsentry/sentry-dotnet/tree/main/samples), [F# Console](https://github.com/sentry-demos/fsharp), and[F# Giraffe](https://github.com/sentry-demos/giraffe)
```

File: src/platforms/dotnet/common/unit-testing.mdx [hunk 7]
```diff
@@ -1,22 +1,43 @@
 ---
-title: Unit Testing
-description: "Learn about unit testing with `ISentryClient` and `IHub`."
+title: Compatibility and Unit Testing
```
Comment (bruno-garcia) [inline src/platforms/dotnet/common/unit-testing.mdx:2]: Not sure it makes much sense to have a page for the two things.
They are quite unrelated.

File: src/platforms/dotnet/common/unit-testing.mdx [hunk 8]
```diff
@@ -1,22 +1,43 @@
...
+- .NET Framework 4.6.1
+- .NET Core 2.0
+- Mono 5.4
+- Xamarin.Android 8.0
+- Xamarin.Mac 3.8
+- Xamarin.iOS 10.14
+- Universal Windows Platform 10.0.16299
 
-The `ISentryClient` exposes the `CaptureEvent` method and its implementation `SentryClient` is responsible for queueing the event to be sent to Sentry. It also abstracts away the internal transport.
+## Unit Testing
```
Comment (bruno-garcia) [inline src/platforms/dotnet/common/unit-testing.mdx:17]: I wouldn't mix the information about where we run our unit tests with the documentation about how to do Unit Testing with our SDK.

This first part is just about giving confidence to users about how compatible our SDK is with different versions and runtimes. The second one is educating users on how they can use Sentry in their code, but still write proper unit tests.

File: src/platforms/dotnet/migration.mdx [hunk 9]
```diff
@@ -66,101 +64,94 @@ catch (Exception e)
...
-#### ASP.NET Core `IncludeRequestPayload`
+To match Sentry's default environment naming convention, unless explicitly set otherwise, the Sentry environment will be reported in lowercase. That means if `ASPNET_ENVIRONMENT` is equal to `Production`, the Sentry SDK will report it as `production`.
+
+### Removal of `IncludeRequestPayload`
 
 The option `IncludeRequestPayload` that was deprecated in the Sentry SDK 2.0 has been removed.
 The replacement is the option `MaxRequestBodySize`.
 
-#### `IHub` and `ISentryClient` Received New Methods
+### New Method for `IHub` and `ISentryClient`
```
Comment (bruno-garcia) [inline src/platforms/dotnet/migration.mdx:97]: I believe it should be then:

```suggestion
### New Methods on `IHub` and `ISentryClient`
```

In other words: we added new methods to the interfaces IHub and ISentryClient

File: src/platforms/dotnet/migration.mdx [hunk 10]
```diff
@@ -214,8 +205,8 @@ Here is a list of features available in Sentry SDK that are not available in Sha
 All packages are:
 
 - Strong named
-- Tested on Windows, Linux and macOS
-- Tested on .NET Core, .NET Framework and Mono
+- Tested on Windows, Linux. and macOS
```
Comment (bruno-garcia) [inline src/platforms/dotnet/migration.mdx:208]:
```suggestion
- Tested on Windows, Linux, and macOS
```

**Review by bruno-garcia — COMMENTED** [review]
I know it's a draft but I thought of leaving some early suggestions

File: src/platforms/dotnet/common/unit-testing.mdx (hunk 7 above)
Comment (PeloWriter) [reply src/platforms/dotnet/common/unit-testing.mdx:2]: Ah, got it. Thanks.

File: src/includes/getting-started-install/dotnet.mdx (hunk 3 above)
Comment (PeloWriter) [reply src/includes/getting-started-install/dotnet.mdx:17]: I will move it. Thanks.

File: src/platforms/dotnet/common/unit-testing.mdx (hunk 8 above)
Comment (PeloWriter) [reply src/platforms/dotnet/common/unit-testing.mdx:17]: Got it. Thanks for clarifying.

**Comment by PeloWriter** [comment]
Based on user input on Discord, we're adding content regarding disabling integrations. Note that the code sample needs careful review @bruno-garcia.

File: src/platforms/dotnet/configuration/disable-integrations.mdx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    /// <summary>
+    /// SentryOptions extensions.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public static class SentryOptionsExtensions
+    {
+        public static void DisableAppDomainUnhandledExceptionCapture(this SentryOptions options) =>
+            options.RemoveIntegration<AppDomainUnhandledExceptionIntegration>();
+    }
+}
```
Comment (bruno-garcia) [inline src/platforms/dotnet/configuration/disable-integrations.mdx:36]:
```suggestion
using Sentry;

// Add it to your Sentry SDK initialization block:
options.DisableAppDomainUnhandledExceptionCapture()
```

File: src/platforms/dotnet/configuration/disable-integrations.mdx [hunk 12]
```diff
@@ -0,0 +1,53 @@
+---
+title: Disable Integrations
+sidebar_order: 200
+description: "Learn more about the methods that disable integrations."
+---
+
+All of Sentry’s SDKs provide integrations, which extend functionality of the SDK.Integrations are enabled by default, though you can disable them if needed.
```
Comment (bruno-garcia) [inline src/platforms/dotnet/configuration/disable-integrations.mdx:7]:
```suggestion
All of Sentry’s SDKs provide integrations, which extend functionality of the SDK. Most integrations are enabled by default, though you can disable them if needed.
```

File: src/platforms/dotnet/configuration/disable-integrations.mdx [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+
+To disable an integration, for eample, the automatic capture of unhandled exceptions:
```
Comment (bruno-garcia) [inline src/platforms/dotnet/configuration/disable-integrations.mdx:9]:
```suggestion
To disable an integration, for example, the automatic capture of unhandled exceptions:
```

File: src/includes/getting-started-primer/dotnet.mdx [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+
+**Features:**
+
+- Multiple integrations with additional features such as Serilog, log4net, ASP.NET Core, EntityFramework, Xamarin.
+- <PlatformLink to="/enriching-events/breadcrumbs/#automatic-breadcrumbs">Breadcrumbs automatically</PlatformLink> captured
+- <PlatformLink to="/enriching-events/attachments/">Attachments</PlatformLink> enrich your event by storing additional files, such as config or log files.
+- <PlatformLink to="/enriching-events/user-feedback/">User Feedback</PlatformLink> provides the ability to collect user information when an event occurs.
+- [Performance Monitoring](/product/performance/) creates transactions to track the performance of your application
+- Automatically captures errors including: unhandled exceptions and unobserved task exceptions.
+- Events [enriched](https://docs.sentry.io/platforms/apple/enriching-events/context/) with device data.
```
Comment (bruno-garcia) [inline src/includes/getting-started-primer/dotnet.mdx:13]:
```suggestion
- Events <PlatformLink to="/enriching-events/context/">enriched</PlatformLink>  with device data.
```

File: src/includes/getting-started-verify/dotnet.mdx [hunk 15]
```diff
@@ -1,13 +1,17 @@
-You can verify Sentry is capturing unhandled exceptions by raising an exception. For example, you can use the following snippet to raise a `NullReferenceException`:
-
 ```csharp
-using (SentrySdk.Init("___PUBLIC_DSN___"))
+using Sentry;
+try
```
Comment (bruno-garcia) [inline src/includes/getting-started-verify/dotnet.mdx:3]:
```suggestion

try
```

File: src/includes/getting-started-primer/dotnet.mdx [hunk 16]
```diff
@@ -0,0 +1,16 @@
+Our .NET SDK supports all recent versions of the platform, and integrates well with a variety of popular frameworks and packages in the ecosystem. It gives developers helpful hints for where and why an error or performance issue might have occurred.
+
+This <PlatformLink to="/compatibility/">SDK is compatible</PlatformLink> with .NET Standard 2.0 and .NET Framework 4.6.1. For older versions, such as .NET Framework 3.5, please refer to our [legacy SDK](/clients/csharp/).
+
+**Features:**
+
+- Multiple integrations with additional features such as Serilog, log4net, ASP.NET Core, EntityFramework, Xamarin.
+- <PlatformLink to="/enriching-events/breadcrumbs/#automatic-breadcrumbs">Breadcrumbs automatically</PlatformLink> captured
```
Comment (bruno-garcia) [inline src/includes/getting-started-primer/dotnet.mdx:8]:
```suggestion
- <PlatformLink to="/enriching-events/breadcrumbs/#automatic-breadcrumbs">Breadcrumbs automatically</PlatformLink> captured.
```

File: src/includes/getting-started-verify/dotnet.mdx [hunk 17]
```diff
@@ -1,13 +1,17 @@
...
+catch (Exception ex)
+{
+    SentrySdk.CaptureException(ex);
+}
 ```
-
 ```fsharp
-use __ = SentrySdk.Init("___PUBLIC_DSN___")
-raise <| NullReferenceException ()
+try
```
Comment (bruno-garcia) [inline src/includes/getting-started-verify/dotnet.mdx:13]:
```suggestion
open Sentry

try
```

File: src/platforms/dotnet/common/compatibility.mdx [hunk 18]
```diff
@@ -0,0 +1,31 @@
+---
+title: Compatibility
+sidebar_order: 50
+description: "Learn more about which version the SDK is compatible with."
+---
+
+According to this [compatibility table](https://docs.microsoft.com/en-us/dotnet/standard/net-standard), the package is compatible with the following versions or newer:
```
Comment (bruno-garcia) [inline src/platforms/dotnet/common/compatibility.mdx:7]:
```suggestion
According to the [compatibility table](https://docs.microsoft.com/en-us/dotnet/standard/net-standard), the package is compatible with the following versions or newer:
```

File: src/platforms/dotnet/common/compatibility.mdx [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+
+<Note>
+
+If you're using an older version of .NET Framework or Mono, [our legacy SDK](/clients/csharp/) supports .NET Framework as early as 3.5.
+
+</Note>
+
+## Unit Testing
+
+We run our unit/integration tests against:
```
Comment (bruno-garcia) [inline src/platforms/dotnet/common/compatibility.mdx:25]: I phrase it like this a few years back on the .NET repo but I wonder if we can reword this a bit nicer?

```suggestion
Sentry validates the SDK functionality by running its test suit in the following platforms:
```

File: src/platforms/dotnet/configuration/disable-integrations.mdx [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+All of Sentry’s SDKs provide integrations, which extend functionality of the SDK. Most integrations are enabled by default, though you can disable them if needed.
+
+To disable an integration, for example, the automatic capture of unhandled exceptions:
+
+```csharp
+using Sentry;
+
+// Add it to your Sentry SDK initialization block:
+options.DisableAppDomainUnhandledExceptionCapture()
+```
```
Comment (bruno-garcia) [inline src/platforms/dotnet/configuration/disable-integrations.mdx:16]: It's pretty much the same but for consistency:

````suggestion
```

```fsharp
open Sentry

// Add it to your Sentry SDK initialization block:
options.DisableAppDomainUnhandledExceptionCapture()
```
````

**Review by bruno-garcia — APPROVED** [review]
My previous suggestion had a copy paste error (links to apple docs) but other than that the rest are just nits.

File: src/includes/getting-started-primer/dotnet.mdx (hunk 16 above)
Comment (PeloWriter) [reply src/includes/getting-started-primer/dotnet.mdx:8]: actually, we don't want periods on bullets. I'll fix as I see this is inconsistent in the list.

File: src/platforms/dotnet/common/compatibility.mdx (hunk 19 above)
Comment (PeloWriter) [reply src/platforms/dotnet/common/compatibility.mdx:25]: How about "Sentry validates our SDK functionality running a test suite against the following platforms"

**Pull request merged**